src/messages.rs

```diff
@@ -267,6 +263,9 @@ impl RoutingMessage {
 #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Debug, RustcEncodable, RustcDecodable)]
 pub struct SignedMessage {
     encoded_body : Vec<u8>,
+    claimant     : Address,
```

Comment:
Apart from the obvious problems that it increases bandwidth and that it will cause additional runtime checks - more lines of code and more CPU hungry (even if only little) - my biggest concern is that to me redundancy indicates bad design. I'll let you two be judges of it though, just wanted you to know about it.